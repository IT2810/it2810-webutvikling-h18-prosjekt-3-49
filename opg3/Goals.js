import React from 'react';
import {View, Text, TextInput} from 'react-native';
import Storage from './Storage';

export default class Goals extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            storage: new Storage(),
            goals: ""
        }
    }

    componentDidMount() {
        this.state.storage._retrieveData("key1").then((res) => {
            this.setState({goals: res});
        });
    }

    render() {
        return (
            <View>
                <Text>
                    Dine mÃ¥l er: {this.state.goals}
                </Text>
                <TextInput
                    onSubmitEditing={(e) => {
                        this.state.storage._appendData("key1", e.nativeEvent.text)
                    }}
                />
            </View>
        )
    }
}