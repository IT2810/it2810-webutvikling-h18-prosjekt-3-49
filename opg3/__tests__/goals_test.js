import React from 'react';
import Goals from '../components/Goals';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Goals />).toJSON();
    expect(tree).toMatchSnapshot();
});