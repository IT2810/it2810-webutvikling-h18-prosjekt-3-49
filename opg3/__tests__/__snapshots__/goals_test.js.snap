// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Dine m√•l er: 
    
  </Text>
  <TextInput
    allowFontScaling={true}
    onSubmitEditing={[Function]}
  />
</View>
`;
